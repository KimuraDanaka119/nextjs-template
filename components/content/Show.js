import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import urljoin from 'url-join';

import withI18N from '../hoc/withI18N';
import withConfig from '../hoc/withConfig';
import Error from '../common/Error';

const Show = ({ config, result }) => {
  if (result.statusCode >= 400) {
    return <Error statusCode={result.statusCode} />;
  }

  const image = result.data.image
    ? result.data.image.medium
    : urljoin(config.paths.images, 'placeholder.png');

  return (
    <div>
      <div className="show">
        <h1>{result.data.name}</h1>
        <div className="row">
          <div className="col-md-3">
            <img className="img-fluid" src={image} />
          </div>
          <div className="col-md-9">
            <div dangerouslySetInnerHTML={{ __html: result.data.summary }}></div>
          </div>
        </div>
        <div className="row text-center">
          <div className="col-md-12">
            <Link href="/index" as="/">
              <a className="btn btn-primary">Back</a>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

Show.propTypes = {
  config: PropTypes.object,
  result: PropTypes.object,
};

export default withI18N(withConfig(Show), ['show', 'error']);
