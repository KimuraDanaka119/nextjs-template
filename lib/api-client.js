import fetch from 'isomorphic-unfetch';
import config from 'configs/config';

export default class ApiClient {
  async doFetch(path) {
    const url = config.api.baseURL + path;

    const res = await fetch(url);
    const data = await res.json();

    return {
      statusCode: res.status,
      data,
    };
  }

  async getShows(keyword) {
    const res = await this.doFetch(`/search/shows?q=${keyword}`);

    return res;
  }

  async getShow(id) {
    const res = await this.doFetch(`/shows/${id}`);

    return res;
  }
}
