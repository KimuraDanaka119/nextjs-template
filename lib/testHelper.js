import ApiClient from './api-client';

function t(key) {
  return key;
}

const config = {
  env: 'test',
  versionHash: 'test',
  paths: {
    static: '/static/test',
    styles: '/static/test/styles',
    images: '/static/test/images',
    locales: '/static/test/locales',
  },
  api: {
    baseURL: '.',
  },
};

const apiClient = new ApiClient('.');

export { t, config, apiClient };
