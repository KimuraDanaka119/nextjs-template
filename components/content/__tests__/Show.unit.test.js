import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import { undecorated as Show } from '../Show';
import ApiClient from '../../../lib/api-client';
import { config } from '../../../lib/testHelper';

const apiClient = new ApiClient('.');

describe('<Show />', () => {
  it('should render correctly with result', async () => {
    const result = await apiClient.getShow('975');

    const footer = shallow(<Show config={config} result={result}/>);
    expect(toJson(footer)).toMatchSnapshot();
  });
});
