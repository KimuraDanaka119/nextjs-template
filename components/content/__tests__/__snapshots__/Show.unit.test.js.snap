// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Show /> should render correctly with result 1`] = `
<div>
  <div
    className="show"
  >
    <h1>
      Batman
    </h1>
    <div
      className="row"
    >
      <div
        className="col-md-3"
      >
        <img
          className="img-fluid"
          src="http://static.tvmaze.com/uploads/images/medium_portrait/6/16463.jpg"
        />
      </div>
      <div
        className="col-md-9"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Wealthy entrepreneur Bruce Wayne and his ward Dick Grayson lead a double life: they are actually crime fighting duo Batman and Robin. A secret Batpole in the Wayne mansion leads to the Batcave, where Police Commissioner Gordon often calls with the latest emergency threatening Gotham City. Racing to the scene of the crime in the Batmobile, Batman and Robin must (with the help of their trusty Bat-utility-belt) thwart the efforts of a variety of master criminals, including Catwoman, Egghead, The Joker, King Tut, The Penguin, and The Riddler.</p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="row text-center"
    >
      <div
        className="col-md-12"
      >
        <Link
          as="/"
          href="/index"
        >
          <a
            className="btn btn-primary"
          >
            Back
          </a>
        </Link>
      </div>
    </div>
  </div>
</div>
`;
